### PR TITLE
Add recipe for google-cloud-aiplatform

### PR DESCRIPTION
**Pull request by parthea**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

File: recipes/google-cloud-aiplatform/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+    Google Vertex AI is an integrated suite of machine learning tools and services for building and using ML models 
+    with AutoML or custom code. It offers both novices and experts the best workbench for the entire machine learning 
+    development lifecycle.
+  doc_url: {{ doc_url }}
+  dev_url: {{ dev_url }}
+
+extra:
+  recipe-maintainers:
+    - parthea
+    - morgandu
```
Comment (parthea) [inline recipes/google-cloud-aiplatform/meta.yaml:67]: @morgandu Please confirm that you'd like to be added as a maintainer (as per checklist)

File: recipes/google-cloud-aiplatform/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+  host:
+    - python >=3.6
+    - pip
+  run:
+    - python >=3.6
+    - google-api-core-grpc >=1.26.0,<3.0.0dev
+    - proto-plus >=1.10.1
+    - packaging >=14.3
+    - google-cloud-storage >=1.32.0,<2.0.0dev
+    - google-cloud-bigquery >=1.15.0,<3.0.0dev
```
Comment (tswast) [inline recipes/google-cloud-aiplatform/meta.yaml:30]: The "extras" in https://github.com/googleapis/python-aiplatform/blob/v1.10.0/setup.py are a bit of a mess, but I think we can consolidate a few of those into a unified `run_constrained` section.

`run_constrained` is the closest we can get to "extras" https://docs.conda.io/projects/conda-build/en/latest/resources/define-metadata.html#run-constrained It says "these are optional, but if you do have them installed, here are the compatible versions"

```suggestion
    - google-cloud-bigquery >=1.15.0,<3.0.0dev
    
  run_constrained:
    - tensorflow >=2.3.0,<=2.7.0
    - pandas >=1.0.0
    - pyarrow >=6.0.1
```

File: recipes/google-cloud-aiplatform/meta.yaml (hunk 1 above)
Comment (morgandu) [reply recipes/google-cloud-aiplatform/meta.yaml:67]: Thanks for the PR, please add me as maintainer.

File: recipes/google-cloud-aiplatform/meta.yaml (hunk 2 above)
Comment (parthea) [reply recipes/google-cloud-aiplatform/meta.yaml:30]: Thanks Tim!

**Review by tswast — APPROVED** [review]

File: recipes/google-cloud-aiplatform/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
+  sha256: {{ sha256 }}
+
+build:
+  number: 0
+  script: {{ PYTHON }} -m pip install . -vv
+
+requirements:
+  host:
+    - python >=3.6
```
Comment (tswast) [inline recipes/google-cloud-aiplatform/meta.yaml:21]:
```suggestion
    - python
```

File: recipes/google-cloud-aiplatform/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+    - pip
+  run:
+    - python >=3.6
```
Comment (tswast) [inline recipes/google-cloud-aiplatform/meta.yaml:24]:
```suggestion
    - python
```

**Comment by parthea** [comment]
@conda-forge/help-python, please could you review/approve?

File: recipes/google-cloud-aiplatform/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,71 @@
+{% set name = "google-cloud-aiplatform" %}
+{% set version = "1.10.0" %}
+{% set sha256 = "89d2936985eaf71a52152fe62bae9d7554dec6948032526b99ec05a049d1ee39" %}
+{% set dev_url = "https://github.com/googleapis/python-aiplatform" %}
+{% set doc_url = "https://cloud.google.com/python/docs/reference/aiplatform/latest" %}
```
Comment (BastianZim) [inline recipes/google-cloud-aiplatform/meta.yaml:5]: If it's just used once or twice it's better to have this hardcoded. Makes it easier for the bot as well. Please add these three directly instead.

File: recipes/google-cloud-aiplatform/meta.yaml [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
+  sha256: {{ sha256 }}
+
+build:
+  number: 0
```
Comment (BastianZim) [inline recipes/google-cloud-aiplatform/meta.yaml:16]: I would ask upstream to remove the empty `scripts` tag and make this noarch as that seems to be the only thing holding it up.

If that's currently not possible you can also just add a patch here removing it.

Ref: https://github.com/googleapis/python-aiplatform/blob/1c341544e9bd94c6ff0ee41177565c8c078673a3/setup.py#L114

Comment (parthea) [reply recipes/google-cloud-aiplatform/meta.yaml:16]: Thanks for this info! I've opened a PR upstream : https://github.com/googleapis/python-aiplatform/issues/1003

File: recipes/google-cloud-aiplatform/meta.yaml (hunk 5 above)
Comment (parthea) [reply recipes/google-cloud-aiplatform/meta.yaml]: Done.

File: recipes/google-cloud-aiplatform/meta.yaml (hunk 6 above)
Comment (BastianZim) [reply recipes/google-cloud-aiplatform/meta.yaml:13]: Great! If you're not sure how soon a new release with that will be cut, you can also include that as a patch here. Let me know if you need any help with that.

**Comment by parthea** [comment]
Duplicate of https://github.com/conda-forge/staged-recipes/pull/18100

**Pull request closed without merging**